src/StorageFlattening.cpp

```diff
@@ -400,13 +478,31 @@ class FlattenDimensions : public IRMutator {
     }
 
     Stmt visit(const For *op) override {
+        for (auto &p : hoisted_storages) {
+            Expr expanded_min = simplify(expand_expr(op->min, scope));
+            Expr expanded_extent = simplify(expand_expr(op->extent, scope));
+            Interval loop_bounds = Interval(expanded_min, expanded_min + expanded_extent - 1);
```

Comment:
Suggest doing this instead, in case expanded_min + expanded_extent can simplify:
```
Expr expanded_min = simplify(expand_expr(op->min, scope));
Expr expanded_extent = expand_expr(op->extent, scope);
Interval loop_bounds = Interval(expanded_min, simplify(expanded_min + expanded_extent - 1));
```